### PR TITLE
add example with for await

### PR DESCRIPTION
**Pull request by caub**
Added an example using asyncIterator (`for await`)

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -13,6 +13,7 @@
     "prefer-template": "error"
   },
   "settings": {
+    "import/parser": "babel-eslint",
```
Comment (shellscape) [inline .eslintrc]: ah yeah this is a bad change. renaming the file is no good and adding additional config to it for the examples isn't worth it. the project doesn't use babel explicitly, so it's best to just ignore the offending example file in .eslintignore.

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 language: node_js
 node_js:
-  - 10
-  - 8
-  - 6
+  - "10"
```
Comment (shellscape) [inline .travis.yml]: please leave CI up to maintainers and revert these changes :)

File: examples/transform.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 const write = require('csv-write-stream')
-const through = require('through2')
```
Comment (shellscape) [inline examples/transform.js]: there's value in showing how to use this with `through2` since that's still widely used. renaming the original `transform.js` to `transform-through2.js` and then modifying this file would've been a better route.

File: package.json [hunk 4]
```diff
@@ -42,16 +42,18 @@
     "@commitlint/cli": "^7.0.0",
     "@commitlint/config-conventional": "^7.0.1",
     "ava": "^0.25.0",
+    "babel-eslint": "^10.0.1",
     "bops": "^0.1.1",
     "chalk": "^2.4.1",
     "concat-stream": "^1.4.5",
     "csv-spectrum": "^1.0.0",
+    "csv-write-stream": "^2.0.0",
```
Comment (shellscape) [inline package.json]: looks like this dependency snuck through

File: package.json [hunk 5]
```diff
@@ -42,16 +42,18 @@
     "@commitlint/cli": "^7.0.0",
     "@commitlint/config-conventional": "^7.0.1",
     "ava": "^0.25.0",
+    "babel-eslint": "^10.0.1",
```
Comment (shellscape) [inline package.json]: please remove. see commit on `.eslintrc.json`

File: package.json [hunk 6]
```diff
@@ -42,16 +42,18 @@
...
     "concat-stream": "^1.4.5",
     "csv-spectrum": "^1.0.0",
+    "csv-write-stream": "^2.0.0",
     "eslint": "^5.4.0",
     "eslint-config-standard": "^11.0.0",
     "eslint-plugin-import": "^2.14.0",
     "eslint-plugin-node": "^7.0.1",
     "eslint-plugin-promise": "^4.0.0",
-    "eslint-plugin-standard": "^3.1.0",
+    "eslint-plugin-standard": "^4.0.0",
```
Comment (shellscape) [inline package.json]: please allow maintainers to manage dependency updates and revert this change

File: .travis.yml (hunk 2 above)
Comment (shellscape) [reply .travis.yml]: understood, but that's still a CI commit on a docs PR for an error that's only being displayed on your end :)

File: examples/transform.js (hunk 3 above)
Comment (shellscape) [reply examples/transform.js]: it doesn't, and we should. however, `through2` is still getting 4 million downloads a month and is still widely used. removing its usage is removing value for users. that's why I suggested renaming the file in its original state to express that it's a through2 example. that way we keep both.

File: package.json (hunk 4 above)
Comment (shellscape) [reply package.json]: ah, you are correct.

**Comment by caub** [comment]
package-lock.json is already committed, it's a file that change when installing a new package, so it's normal that it changed

asyncIterator needs no flag on node10 (LTS)

I appreciate that people like try to maintain packages, but I regret how inefficient the process is currently, concerning also `through2` vs a builtin without added complexity, it's very logic to opt for the latter, whatever the number of download of the former, time changes

I don't think this commit will make it, too bad, have a good week, will close some time later, and avoid contributing here later

**Comment by shellscape** [comment]
No worries. I'm sorry you found the process to be inefficient, and regrettable on your decision. If you'd like to reopen later, please let us know.
